### DISCUSSION TITLE
Storage pool file permissions

### DISCUSSION THREAD

**Discussion by sli720** [vagrant-libvirt/vagrant-libvirt#1313] [Q&A] [answered]
Hello,

I have a question about the file permissions of the images within a storage pool. I created my own storage pool and used it within the Vagrantfile through the variable storage_pool_name. 

config.vm.provider :libvirt do |libvirt|
    libvirt.storage_pool_name = "ci_images"
    libvirt.storage :file, :size => '100G', :path => 'pulp_storage.img', :device => 'vdb'
    libvirt.storage :file, :size => '10G', :path => 'pulp_pgsql.img', :device => 'vdc'
  end

All images are  then saved with 744 permissions except the main image (pulp_pulp.img):

-rwxr--r--  1 libvirt-qemu kvm  1735852032 Jun 19 11:04 centos-VAGRANTSLASH-stream8_vagrant_box_image_20210210.0.img
-rwxr--r--  1 libvirt-qemu kvm      196768 Jun 19 11:44 pulp_pgsql.img
-rw-------  1 libvirt-qemu kvm      393216 Jun 19 11:44 pulp_pulp.img
-rwxr--r--  1 libvirt-qemu kvm      198208 Jun 19 11:44 pulp_storage.img

I'm asking because I want to copy these images as a normal user and I don't understand where the 600 for the pulp_pulp.img comes from and where I can configure it.

**Comment by electrofelix** [comment, marked as answer]
Partly the 744 is due to accepting a default setting in fog-libvirt that I think is questionable security wise.

To copy the files as a user I'd recommend use of `virsh vol-download` because it will know whether you have access to the storage, better generally to use the APIs provided that stepping behind just because you can see the storage directly. That way it should also work irrespective of whether the libvirt is local or remote, user or system, if you have access via libvirt it'll work.

I guess we should make it configurable while defaulting to the most secure setting, that way individuals can decide to accept the less secure model rather having what was imposed by the templates

**Reply by sli720** [reply]
Thanks for the answer. But by default vagrant/libvirt runs as root so the pools are not accessible from other user accounts. Is it possible to tell in the Vagrantfile to use not root for creating libvirt vms/images?

**Reply by electrofelix** [reply]
Vagrant-libvirt runs as the user that runs it, not sure why you would run it as root. The pools are accessible from the same user that runs vagrant, because if they aren't they won't be accessible to vagrant-libvirt.

Maybe your environment is setup so that you need to specify the right connection to virsh
```
virsh -c qemu:///system pool-list
```

It's not possible to currently control the permissions, but if you can create a VM in libvirt using the Vagrantfile, then you will be able to access all the same storage as the same user unless you are running vagrant via sudo.

**Reply by sli720** [reply]
Yes it runs as my user but for some reason it only looks for pools created by the root user. When I create a pool with my user it can not be found through storage_pool_name and I don't understand what I'm doing wrong here.

**Reply by electrofelix** [reply]
It's probably the connection URL, it's likely when running virt-manager or virsh as a user it is defaulting to using `qemu:///session` and when run as root it uses `qemu:///system`. You can override this as a user by passing in the latter as an argument to virsh followed by the virsh command you want to run
```
virsh -c qemu:///system <command>
```

If you are using a new enough release of vagrant-libvirt it should pick up the `LIBVIRT_DEFAULT_URI` is set of some distros that causes this behaviour difference instead of defaulting to the system profile. This should make it more obvious when it needs to be set explicitly.

**Reply by sli720** [reply]
Yes this works. Thank you very much!
